lib/perl/Genome/Model/Tools/Picard/MarkDuplicates.pm

```diff
@@ -121,15 +140,11 @@ sub execute {
         }
     }
     $dedup_cmd .= ' ' . $self->get_max_filehandles_param;
+}
 
-
-    $self->run_java_vm(
-        cmd => $dedup_cmd,
-        input_files => [$self->input_file],
-        output_files => [$self->output_file, $self->metrics_file],
-        skip_if_output_is_present => 0,
-    );
-    return 1;
+sub build_cmdline_list {
+    my $self = shift;
+    return split(' ', $self->_java_cmdline($self->build_cmdline_string));
```

Comment:
Same here, might be safer to use [Test::ParseWords::shellwords](http://perldoc.perl.org/Text/ParseWords.html).